### DISCUSSION TITLE
Detection of weasels

### DISCUSSION THREAD

**Discussion by bartbutenaers** [blakeblackshear/frigate#19794] [Ask A Question] [answered]
Hi everybody,

I was wondering if it is possible to detect weasels/martens?  Or perhaps I can use Frigate+ for this?

Thanks!
Bart

**Comment by inkeep-ai-assistant** [comment]
Frigate’s default (free) object detection models do not include weasel or marten as recognized labels—these models focus on more common animals like birds, cats, and dogs[(1)](https://github.com/blakeblackshear/frigate/discussions/18797). Frigate+ expands the list of detectable objects and does include additional animals such as deer[(2)](https://github.com/blakeblackshear/frigate-hass-integration/issues/901), but there is no explicit mention in the available documentation or supported label types that weasel or marten are currently supported as direct detection labels in Frigate+[(1)](https://github.com/blakeblackshear/frigate/discussions/18797).

If you require detection of a specific animal not on the standard label list, Frigate+ offers a fine-tuning workflow where users can train custom models with annotated images; however, this process is more involved and may not be available for every possible animal without explicit support[(1)](https://github.com/blakeblackshear/frigate/discussions/18797).

📚 **Sources:**
1. [Is it possible to track mice or other animals using frigate ?](https://github.com/blakeblackshear/frigate/discussions/18797)
2. [Deer not recognized? Is animal detection possible?](https://github.com/blakeblackshear/frigate-hass-integration/issues/901)

---

Have another question? Just tag [@inkeep](https://inkeep.com).

**Comment by bartbutenaers** [comment]
@inkeep
Ok suppose I use Frigate+.  How can I train the model for weasels?  Because currently my Frigate does not detect them, which means I have no recordings that contain images of weasels in my garden.  So I have no images to upload, to train the model for a new animal type from scratch?

**Reply by blakeblackshear** [reply, marked as answer]
I already have mustelid/weasel on my list of objects to add to Frigate+

**Reply by bartbutenaers** [reply]
Ok thanks @blakeblackshear for the fast response!!!

Then I will have a look to switch to Frigate+ 

Without being pushy, is it possible to provide a rough estimate whether that planned is in the near future?
  
Just asking because those weasels are behaving quite badly.  Would be really helpfull to see what is going during the night...

**Reply by blakeblackshear** [reply]
Shorter term would be support for it as a candidate label, but not in the model yet. Adding it in the model requires enough training data to get good performance, and I can't really say how long that would take. In short, it's going to be longer than you would like.

**Comment by inkeep-ai-assistant** [comment]
Frigate+ only supports training on a specific set of labels, and "weasel" is not currently among the supported or candidate label types for annotation or fine-tuning[(1)](https://docs.frigate.video/plus). You cannot add your own custom labels to Frigate+ at this time[(1)](https://docs.frigate.video/plus)[(2)](https://frigate.video/plus). Training the model for a new animal type from scratch (such as weasel) is not possible unless it is added as a supported or candidate label in a future release.

If you want to contribute to faster support for new animal types, you can enable and use candidate labels (where available) to help Frigate+ gather enough data, but "weasel" is not listed among the current candidate labels[(1)](https://docs.frigate.video/plus). Without the ability to label weasel in Frigate+, there is no workflow to train the model specifically for that animal.

📚 **Sources:**
1. [Models](https://docs.frigate.video/plus)
2. [Frigate+](https://frigate.video/plus)



---

Have another question? Just tag [@inkeep](https://inkeep.com).

**Comment by gfi63** [comment]
I had a weasel detected as a cat tonight also, I do also not yet see it as a candidate label, but how could I upload such footage for future training?
Would it be an idea to have a candidate label like other that uploads for future reference? I have now made exports of the video, so if that can be of use, just let me know.

Thanks Gerrit
<img width="640" height="480" alt="reolink_cam_3_cat" src="https://github.com/user-attachments/assets/7e1fc961-0ca5-4720-9484-f41f6b41b3d0" />

**Reply by blakeblackshear** [reply]
Weasels are mustelids, so you should enable that candidate label and use it for weasels.
